### DISCUSSION TITLE
PHP in JS e outras linguagens

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#923] [Discussão]
Estava vendo umas bibliotecas de JS aqui, que permitem escrever PHP no Javascript, e também C,Ruby e Python.

O que acham disso?

http://www.php-js.com/
http://locutus.io/2016/05/announcing-locutus/

**Comment by felipemfp** [comment]
![image](https://user-images.githubusercontent.com/8146112/35781065-c533e4b6-09c3-11e8-9d2b-dddb51092fa0.png)

**Comment by luizbills** [comment]
@rubenmarcus Tem o https://github.com/phpv8/v8js também. É bem util quando algo que vc precisa só tem para nodejs ou para fazer SSR de react pelo php (apenas exemplo). Já ouvi dizer que a comunidade do Laravel usa para fazer SSR de códigos com vue.js

E @felipemfp não seja fechado à ideias que não te agradam. Justifique seu ponto ou dê sugestões de abordagens melhores.

**Comment by felipemfp** [comment]
> E @felipemfp não seja fechado à ideias que não te agradam. Justifique seu ponto ou dê sugestões de abordagens melhores.

@luizbills, tô só brincando (afinal é domingo :beach_umbrella:). Não tenho experiência para convencer, mas sou do time de backend separado de frontend (#899) (ou sem backend at all, _serveless_). Também não vejo vantagens em misturar php (nada contra :v:) com JavaScript (se for SSR, funciona melhor com nodejs).

Mas se for para integrar em um sistema já existente, pode valer a pena. Mas se for projeto novo, investir numa arquitetura bem definida pode ser melhor do que misturar as coisas (APIzinha com Lumen e um webapp com algum framework spa).

~E não sou fechado às ideas que não me agradam, tanto que estou seguindo a thread~ :smile:

**Comment by bernardodiasc** [comment]
Esse tipo de coisa não é feita pra ser adotada sem razão, não é "trendy" e nem foi feito pra ser, quem precisa usar esse tipo de lib sabe muito bem o motivo e os casos de uso podem ser bastante específicos. Ao meu vez um caso de uso onde isso salvaria a pele é pra integração de libs feitas em diferentes linguagens, como o @luizbills comentou.
